### PR TITLE
Provider 503 expose carrier server status in brand api

### PR DESCRIPTION
**Pull request by ironBatista**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 

#### Description
Created custom endpoint for carrier_server, for status retrieval.

### CODE REVIEW

**Comment by ironArt3mis** [comment]
Rebase completed.

File: library/Ivoz/Kam/Infrastructure/Kamailio/TrunksClient.php [hunk 1]
```diff
@@ -113,6 +113,7 @@ public function getLcrGatewayInfo($gw_id): array
             return [];
         }
 
+
```
Comment (mmadariaga) [inline library/Ivoz/Kam/Infrastructure/Kamailio/TrunksClient.php:116]: revert this change

File: library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                $carrierServerDto
+                    ->getId()
+            );
+
+        if (is_null($kamTrunksLcrGateway)) {
+            return $carrierServerDto;
+        }
+
+        try {
+            $dumpGw = $this->trunksClient->getLcrGatewayInfo(
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Application/Service/CarrierServer/CarrierServerDtoAssembler.php:55]: $lcrGatewayInfo would be a better name

File: library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php [hunk 3]
```diff
@@ -14,6 +19,27 @@ class CarrierServerDto extends CarrierServerDtoAbstract
...
+            $baseAttributes = [
+                'id' => 'id',
+                'ip' => 'ip',
+                'hostname' => 'hostname',
+                'sipProxy' => 'sipProxy',
+                'authNeeded' => 'authNeeded',
+                'status' => [[
+                    'registered',
+                    'inProgress',
+                    'expires'
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerDto.php:32]: this attributes do not exist in CarrierServerRegistrationStatus

File: library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerRegistrationStatus.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    private $registered = false;
+
+    public function __construct(
+        int $statusCode
+    ) {
+        if ($statusCode === 0) {
+            $this
+                ->setRegistered(true);
+        }
```
Comment (mmadariaga) [inline library/Ivoz/Provider/Domain/Model/CarrierServer/CarrierServerRegistrationStatus.php:19]: ``
```suggestion
            $this->setRegistered(
                $statusCode === 0
            );
```

File: web/rest/brand/config/api/raw/provider.yml [hunk 5]
```diff
@@ -770,6 +770,18 @@ Ivoz\Provider\Domain\Model\Carrier\Carrier:
...
+    put: ~
+    delete: ~
+    get_status_item:
+      method: 'GET'
+      path: '/carrier_servers/{id}/status'
+      operation_normalization_context: !php/const Ivoz\Provider\Domain\Model\CarrierServer\CarrierServerDto::CONTEXT_STATUS
+      normalization_context:
+        groups: [
+          !php/const Ivoz\Provider\Domain\Model\CarrierServer\CarrierServerDto::CONTEXT_STATUS
+        ]
```
Comment (mmadariaga) [inline web/rest/brand/config/api/raw/provider.yml:784]: Missing API tests

**Review by mmadariaga — CHANGES_REQUESTED** [review]

**Review by mmadariaga — APPROVED** [review]

**Comment by ironArt3mis** [comment]
Rebase completed.

**Comment by ironArt3mis** [comment]
Rebase completed.

**Pull request merged**